src/riak_kv_vnode.erl

```diff
@@ -1486,8 +1486,8 @@ prepare_put(State=#state{vnodeid=VId,
     %% old object to know how the indexes have changed.
     IndexBackend = is_indexed_backend(Mod, Bucket, ModState),
     IsSearchable = ?IS_SEARCH_ENABLED_FOR_BUCKET(BProps),
-    RequiresReadBeforeWrite = LWW andalso (not IndexBackend) andalso (not IsSearchable),
-    case RequiresReadBeforeWrite of
+    SkipReqdBeforeWrite = LWW andalso (not IndexBackend) andalso (not IsSearchable),
```

Comment:
This appears to address my comment on the previous commit. Assuming this should be SkipReadBeforeWrite.